src/lib/backend.c

```diff
@@ -187,14 +181,35 @@ CK_RV backend_get_tokens(token **tok, size_t *len) {
                 return rv;
             } else {
                 LOGW(msg);
-                rv = CKR_OK;
             }
         }
+        LOGV("FAPI + Esysdb returned %zi token", *len);
     }
 
-    if (esysdb_init) {
-        (*tok)[*len] = tmp;
-        *len += 1;
+    /* -1 for starting at id 1 and -1 for the empty token */
+    if (*len >= MAX_TOKEN_CNT - 2) {
```

Comment:
This would force current users have MAX tokens from esysdb into an error state, very unlikely, I've never seen more than a handful active at a time. But the old esysdb unitialized token logic had the same bug, it tried to always add an uninitialized token if one wasn't present, and if at max count, failed.

Reply:
It will deny current users with MAX tokens a new empty token; existing tokens are usable if they were usable before.

Reply:
oh yeah, duh, LOGW + CKR_OK